test/integration/integration_test.go

```diff
@@ -979,29 +1097,184 @@ func TestDockerfileBuildSourceScriptsRun(t *testing.T) {
+	defer os.RemoveAll(tempdir)
+
+	sourcecode := filepath.Join(tempdir, "sourcecode")
+	sourcescripts := filepath.Join(sourcecode, ".s2i", "bin")
+	err = os.MkdirAll(sourcescripts, 0777)
+	if err != nil {
+		t.Errorf("Unable to create injection dir: %v", err)
+	}
+
+	assemble := filepath.Join(sourcescripts, "assemble")
+	_, err = os.OpenFile(assemble, os.O_RDONLY|os.O_CREATE, 0666)
+	if err != nil {
+		t.Errorf("Unable to create assemble file: %v", err)
+	}
+
+	config := &api.Config{
+		BuilderImage:    "docker.io/centos/nodejs-8-centos7",
+		AssembleUser:    "",
+		ImageWorkDir:    "",
+		Source:          git.MustParse("file:///" + filepath.ToSlash(sourcecode)),
+		ForceCopy:       true,
+		ImageScriptsURL: "image:///usr/custom/s2i",
+		Injections:      api.VolumeList{},
+		Destination:     "/destination",
+
+		Environment: api.EnvironmentList{},
+		Labels:      map[string]string{},
+
+		AsDockerfile: filepath.Join(tempdir, "Dockerfile"),
+	}
+	expected := []string{
+		"COPY --chown=1001:0 upload/scripts /destination/scripts",
+		"RUN /destination/scripts/assemble",
+		"CMD /usr/custom/s2i/run",
+	}
+	expectedFiles := []string{
+		filepath.Join(tempdir, "upload/scripts/assemble"),
+	}
+	runDockerfileTest(t, config, expected, nil, expectedFiles)
+}
+
+func TestDockerfileBuildScriptsAndImageURL(t *testing.T) {
 	tempdir, err := ioutil.TempDir("", "s2i-dockerfiletest-dir")
 	if err != nil {
 		t.Errorf("Unable to create temporary directory: %v", err)
 	}
 	defer os.RemoveAll(tempdir)
 
+	assemble := filepath.Join(tempdir, "assemble")
+	_, err = os.OpenFile(assemble, os.O_RDONLY|os.O_CREATE, 0666)
+	if err != nil {
+		t.Errorf("Unable to create assemble file: %v", err)
+	}
+
 	config := &api.Config{
 		BuilderImage:    "docker.io/centos/nodejs-8-centos7",
 		AssembleUser:    "",
 		ImageWorkDir:    "",
-		ImageScriptsDir: "/usr/libexec/s2i",
-		Incremental:     true,
 		Source:          git.MustParse("https://github.com/sclorg/nodejs-ex"),
-		ScriptsURL:      "",
-		Tag:             "test:tag",
+		ScriptsURL:      "file://" + filepath.ToSlash(tempdir),
+		ImageScriptsURL: "image:///usr/some/dir",
 		Injections:      api.VolumeList{},
-		Destination:     "",
+		Destination:     "/destination",
+
+		Environment: api.EnvironmentList{},
+		Labels:      map[string]string{},
+
+		AsDockerfile: filepath.Join(tempdir, "Dockerfile"),
+	}
+	expected := []string{
+		"COPY --chown=1001:0 upload/scripts /destination/scripts",
+		"RUN /destination/scripts/assemble",
+		"CMD /usr/some/dir/run",
+	}
+	expectedFiles := []string{
+		filepath.Join(tempdir, "upload/src/server.js"),
+		filepath.Join(tempdir, "upload/scripts/assemble"),
+	}
+	runDockerfileTest(t, config, expected, nil, expectedFiles)
+}
+
+// TestDockerfileBuildScriptsAndImageURLConflicts tests if both
+// the ScriptsURL and ImageScriptsURL point to a non-image directory.
+// In this event, the ScriptsURL value should take precedence.
+func TestDockerfileBuildScriptsAndImageURLConflicts(t *testing.T) {
```

Comment:
Test here if `ScriptsURL` and `ImageScriptsURL` point to non-image directories. Verifies that `ScriptsURL` overrides `ImageScriptsURL`